### DISCUSSION TITLE
How to Activate the dark and light mode through the registry

### DISCUSSION THREAD

**Discussion by Alhlhli** [moudey/Shell#297] [Q&A]
Hello
I noticed several variables to modify the registry values, I tried and it didn't work
You need to change two values ​​from 0 to 1

<img width="513" alt="image" src="https://github.com/moudey/Shell/assets/134859377/f8c4c5aa-116a-4339-abc3-19df90536ce7">

Attempts you wrote

reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', "AppsUseLightTheme", 1, reg.dword)
reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', "SystemUsesLightTheme", 1, reg.dword)

		item(title='light mod' image=\uE095 reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', "AppsUseLightTheme", 1 reg.dword reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', "SystemUsesLightTheme", 1, reg.dword))
		item(title='Dark mod' image=\uE094 reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', "AppsUseLightTheme", 0, reg.dword)reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', "SystemUsesLightTheme", 0, reg.dword))

		item(title='light mod' image=\uE092 reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', "SystemUsesLightTheme", 1, reg.dword)
		item(title='Dark mod' image=\uE092 reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', "SystemUsesLightTheme", 0, reg.dword)

**Comment by Alhlhli** [comment]
item(title='light mod' image=\uE092 reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', "SystemUsesLightTheme", 1,))
	item(title='Dark mod' image=\uE092 reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', "SystemUsesLightTheme", 0,))

**Comment by Alhlhli** [comment]
I don't know it might be the = sign or the parenthesis

item(title='light mod' image=\uE092)
		(
			cmd=reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', 'SystemUsesLightTheme', 0, reg.dword)
		)

**Comment by RubicBG** [comment]
I'll try to help, but first, which version are you using?

**Reply by Alhlhli** [reply]
1.8.1

**Comment by RubicBG** [comment]
1.  the correct syntax is cmd=reg.set()

```
item(title='Light Mode' image=\uE092 cmd=reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', "SystemUsesLightTheme", 1,))
item(title='Dark Mode' image=\uE092 cmd=reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', "SystemUsesLightTheme", 0,))
```

2. There are some minor things in version 1.8.1 that don't work properly with the reg function - better use one of the beta versions from [discord](https://discord.com/channels/1106387012707168318/1112126326665650176)

3. here is an improved version

```
$is_Light = reg.get('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', 'SystemUsesLightTheme')==1  // !sys.dark
item(title='Change to @if(is_Light, 'Dark', 'Light') Mode' image=\uE092 
    cmd=reg.set('HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize', "SystemUsesLightTheme", if(is_Light,0,1), reg.dword) & command.restart_explorer)
```

**Reply by Alhlhli** [reply]
Unfortunately it didn't work for me

I took a brief idea


	menu(title="Theme" image=\uE26B)
	{
		item(title='Light' image=\uE137 cmd='Light')
		item(title='Dark' image=\uE138 cmd='Dark')
		
	}

And create files Light.cmd $Dark.cmd 
then copy 2 Files to windows directory

`@echo off
cd %~dp0
cd /d "%~dp0"
:: BatchGotAdmin================================================================
:: Amer Alhlhli

REM  --> Check for permissions
>nul 2>&1 "%SYSTEMROOT%\system32\cacls.exe" "%SYSTEMROOT%\system32\config\system"

REM --> If error flag set, we do not have admin.
if '%errorlevel%' NEQ '0' (
    echo Requesting administrative privileges...
    goto UACPrompt
) else ( goto gotAdmin )

:UACPrompt
    echo Set UAC = CreateObject^("Shell.Application"^) > "%temp%\getadmin.vbs"
    echo UAC.ShellExecute "%~s0", "", "", "runas", 1 >> "%temp%\getadmin.vbs"

    "%temp%\getadmin.vbs"
    exit /B

:gotAdmin
    if exist "%temp%\getadmin.vbs" ( del "%temp%\getadmin.vbs" )
    pushd "%CD%"
    CD /D "%~dp0"

:: Start========================================================================

Reg.exe add "HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize" /v "AppsUseLightTheme" /t REG_DWORD /d "0" /f
Reg.exe add "HKCU\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize" /v "SystemUsesLightTheme" /t REG_DWORD /d "0" /f

:: End========================================================================
taskkill /f /im explorer.exe  & start explorer.exe
`


I would like it to be done directly without the CMD files

**Reply by moudey** [reply]
This function `reg.get`  is only in the debug version
